### PR TITLE
add a meson.build file, tests and update the README.MD

### PR DESCRIPTION
**Pull request by peter-urban**
This adds a meson.build file (see issue https://github.com/TEOS-10/GSW-C/issues/72)
I also added a related test matrix updated the README.MD to report how to build the project using meson.

The build file allows installing and testing the project using different configurations.
When installing, it also installs a package config file which should ease integrating with e.g. CMAKE 
However I didn't actually test the package config file (Happy to support you if you run into problems though)

The build file also allows for directly integrating the project as subproject into existing meson 
projects (this is what I use it for at the moment)

Happy to receive feedback on this PR, I'd also be happy to support maintaining this file.
Cheers

### CODE REVIEW

**Review by efiring — CHANGES_REQUESTED** [review]
I'm not very experienced with github actions or with meson, so I defer to @ocefpaf for any detailed checking.  It all looks reasonable, and the tests pass (a lot of tests!).

In addition to the changes you made to README, the meson additions should be added to the file list in README-developer.

File: .github/workflows/test-alternative-compilers.yml
[the remark this answers is not in the document]
Comment (ocefpaf) [reply .github/workflows/test-alternative-compilers.yml:1]: Eric, are you OK if we use this opportunity to rename it to `main`?

**Comment by ocefpaf** [comment]
> I'm not very experienced with github actions or with meson, so I defer to @ocefpaf for any detailed checking. It all looks reasonable, and the tests pass (a lot of tests!).

LGTM. Indeed the test matrix got quite big. I wonder if we could do both shared and static in one go... But then it would be hard to find things in the logs, so probably not worth it. (And this runs quite fast, probably fine as-is.)

File: .github/workflows/test-alternative-compilers.yml
[the remark this answers is not in the document]
Comment (efiring) [reply .github/workflows/test-alternative-compilers.yml:1]: Yes.

File: .github/workflows/test-alternative-compilers.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: run gsw_check (alternative compilers)
 on:
   pull_request:
   push:
-    #branches: [master]
+    branches: [master]
```
Comment (ocefpaf) [inline .github/workflows/test-alternative-compilers.yml:6]:
```suggestion
    branches: [main]
```

File: .github/workflows/tests-meson.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+name: test meson wrapper
+
+on:
+  pull_request:
+  push:
+    branches: [master]
```
Comment (ocefpaf) [inline .github/workflows/tests-meson.yml:6]:
```suggestion
    branches: [main]
```

File: .github/workflows/test-alternative-compilers.yml
[the remark this answers is not in the document]
Comment (peter-urban) [reply .github/workflows/test-alternative-compilers.yml:1]: ok, I adapted the new main branch name in all ci-workflow files

File: .github/workflows/test-alternative-compilers.yml (hunk 1 above)
Comment (peter-urban) [reply .github/workflows/test-alternative-compilers.yml]: adapted the new branch name in all workflows

**Comment by peter-urban** [comment]
> I'm not very experienced with github actions or with meson, so I defer to @ocefpaf for any detailed checking. It all looks reasonable, and the tests pass (a lot of tests!).
> 
> In addition to the changes you made to README, the meson additions should be added to the file list in README-developer.

I added meson.build and meson.options to the README and README-developer file manifest.

**Review by ocefpaf — APPROVED** [review]

**Review by efiring — APPROVED** [review]

**Pull request merged**